### PR TITLE
bowtie2: fix build on aarch64

### PR DESCRIPTION
**Pull request by natsukium**
###### Description of changes

- enable to build on aarch64
- enable passthru.tests
- update the license from gpl3 to gpl3Plus
- add a changelog to the meta

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 227090` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bowtie2</li>
  </ul>
</details>

File: pkgs/applications/science/biology/bowtie2/default.nix [hunk 1]
```diff
@@ -8,19 +8,37 @@ stdenv.mkDerivation rec {
...
+
   meta = with lib; {
     description = "An ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences";
-    license = licenses.gpl3;
+    license = licenses.gpl3Plus;
     homepage = "http://bowtie-bio.sf.net/bowtie2";
+    changelog = "https://github.com/BenLangmead/bowtie2/releases/tag/v${version}";
     maintainers = with maintainers; [ rybern ];
     platforms = platforms.all;
-    broken = stdenv.isAarch64; # only x86 is supported
```
Comment (kirillrdy) [inline pkgs/applications/science/biology/bowtie2/default.nix:24]: doesn't build on aarch64-linux, maybe mark as broken on `stdenv.isAarch64 && stdenv.isLinux` ?

Comment (natsukium) [reply pkgs/applications/science/biology/bowtie2/default.nix:24]: Thanks for your review!
I don't remember much about this PR. I'll check it later.

Comment (kirillrdy) [reply pkgs/applications/science/biology/bowtie2/default.nix:24]: no rush :+1:

Comment (natsukium) [reply pkgs/applications/science/biology/bowtie2/default.nix:24]: We can now build on aarch64-linux.

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 227090` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bowtie2</li>
  </ul>
</details>

**Pull request merged**